File: dlna-dmc/src/main/java/com/android/cast/dlna/ICast.java
package com.android.cast.dlna;

import androidx.annotation.NonNull;

public interface ICast {

    @NonNull
    String getId();

    @NonNull
    String getUri();

    String getName();

    interface ICastVideo extends ICast {

        /**
         * @return video duration, ms
         */
        long getDurationMillSeconds();

        long getSize();

        long getBitrate();
    }
}

File: dlna-core/src/main/java/com/android/cast/dlna/core/ICast.kt
package com.android.cast.dlna.core

interface ICast {
    val id: String
    val uri: String
    val name: String?

    interface ICastVideo : ICast {
        val durationMillSeconds: Long // video duration, ms
        val size: Long
        val bitrate: Long
    }

    interface ICastAudio : ICast {
        val durationMillSeconds: Long // audio duration, ms
        val size: Long
    }

    interface ICastImage : ICast {
        val size: Long
    }
}